include/wx/event.h

```diff
@@ -144,9 +144,43 @@ inline wxEventFunction wxEventFunctionCast(void (wxEvtHandler::*func)(T&))
     wxGCC_WARNING_RESTORE_CAST_FUNCTION_TYPE()
 }
 
+// Special hack to remove "noexcept" from the function type when using C++17 or
+// later: static_cast<> below would fail to cast a member function pointer to a
+// member of a derived class to the base class if noexcept is specified for the
+// former, so remove it first if necessary.
+#if __cplusplus >= 201703L
```

Comment:
I suppose there is a reason why ` || wxCHECK_VISUALC_VERSION(14)` is missing here?

Reply:
AFAIK it doesn't have any problems with casting `noexcept` away, so I prefer to not change anything for it.